src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectcreation/model/ProjectCreationModel.kt

```diff
@@ -21,50 +22,81 @@ class ProjectCreationModel {
     )
     var sourceLanguageProperty: Language by property()
     var targetLanguageProperty: Language by property()
-    var selectedResource: Collection by property()
-    var selectedAnthology: Collection by property()
-    var selectedBook: Collection by property()
-    var anthologyList: ObservableList<Collection> by property(FXCollections.observableArrayList())
-    var bookList: ObservableList<Collection> by property(FXCollections.observableArrayList())
-
+    var collectionList: ObservableList<Collection> = FXCollections.observableArrayList()
     val languages: ObservableList<Language> = FXCollections.observableArrayList()
-    val resources: ObservableList<Collection> = FXCollections.observableArrayList()
+    var collectionStore: ArrayList<List<Collection>> = ArrayList()
+    var creationDepth: Int = 1
+    var bookDepthReached = false
 
     init {
         creationUseCase.getAllLanguages()
                 .observeOnFx()
                 .subscribe { retrieved ->
                     languages.setAll(retrieved)
                 }
+    }
 
+    fun getSourceRepos() {
         creationUseCase.getSourceRepos()
                 .observeOnFx()
                 .subscribe { retrieved ->
-                    resources.setAll(retrieved)
+                    collectionStore.add(retrieved.filter {
+                        it.resourceContainer?.language == sourceLanguageProperty
+                    })
+                    collectionList.setAll(collectionStore[collectionStore.size - 1])
                 }
     }
 
-    fun getResourceChildren() {
-        creationUseCase.getResourceChildren(selectedResource)
-                .observeOnFx()
-                .doOnSuccess {
-                    anthologyList.setAll(it.sortedBy { it.sort })
-                }
-                .subscribe()
+    fun checkLevel(selectedCollection: Collection, workspace: Workspace) {
+        if (collectionList[0].labelKey == "book") {
```

Comment:
Maybe check the selectedCollection directly (`if (selectedCollection.labelKey == "book")`)